### PR TITLE
Updating menu

### PR DESCRIPTION
**Pull request by Duncid**
Making use of the new DropdownMenu.

## WARNING

- Needs to be fully tested, I could not the test the Workspace picker
- I removed things that I did not fully understand -_-'
- eslint is catching an error that I do not understand (and maybe pre-dates my changes?)

### CODE REVIEW

File: front/package-lock.json [hunk 1]
```diff
@@ -925,6 +928,230 @@
         "get-tsconfig": "^4.4.0"
       }
     },
+    "node_modules/@esbuild/android-arm": {
```
Comment (spolu) [inline front/package-lock.json:931]: Let's clean that up

**Review by spolu — COMMENTED** [review]
package-lock.json needs clean-up :)

Will test it to confirm this is :+1: 

Sparkle 0.42 is merged to main?

**Comment by spolu** [comment]
Is this already published?

**Comment by spolu** [comment]
(If so let's bump the version)

**Comment by spolu** [comment]
Will wait for sparkle to be fully merged and publish to test and final review :+1:

**Comment by Duncid** [comment]
> Sparkle 0.42 is merged to main?

No, it's [here](https://github.com/dust-tt/dust/pull/1228), but it's already deployed (so can be merged separately.

File: front/package-lock.json (hunk 1 above)
Comment (Duncid) [reply front/package-lock.json]: Done

**Comment by Duncid** [comment]
FYI, keeps failing with that error that i don't understand (As I don't think I changed that behavior):


./components/sparkle/AppLayout.tsx
69:25  Error: Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator.  @typescript-eslint/no-floating-promises

**Comment by spolu** [comment]
You need to add a `void` before calling async functions that you don't await. Will comment in code :+1:

File: front/components/WorkspacePicker.tsx [hunk 2]
```diff
@@ -15,54 +13,22 @@ export default function WorkspacePicker({
...
-          </div>
-        </Menu.Items>
-      </Menu>
-    </div>
+      <DropdownMenu.Items>
+        {user.workspaces.map((w) => {
+          return (
+            <DropdownMenu.Item
+              key={w.sId}
+              onClick={() => onWorkspaceUpdate(w)}
```
Comment (spolu) [inline front/components/WorkspacePicker.tsx:26]: onWorkspaceUpdate is likely async but we don't await it we need to explicitely say that's fine:
```
() => {
  void onWorskpaceUpdate(w);
}
```

Comment (spolu) [reply front/components/WorkspacePicker.tsx:26]: Due to your onClick handler not being async itself I presume while the previous one was...

Comment (spolu) [reply front/components/WorkspacePicker.tsx:26]: assuming this is the source of error might be in the other file will comment there as well

File: front/components/sparkle/AppLayout.tsx [hunk 3]
```diff
@@ -43,54 +49,31 @@ function NavigationBar({
...
+                          : "https://gravatar.com/avatar/anonymous?d=mp"
+                      }
+                      alt=""
+                    />
+                  </DropdownMenu.Button>
+                  <DropdownMenu.Items origin="topRight">
+                    <DropdownMenu.Item
+                      label="Sign&nbsp;out"
+                      onClick={() => {
+                        signOut({
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/components/sparkle/AppLayout.tsx:69]: Likely this one

**Comment by Duncid** [comment]
All check passed :)

**Comment by Duncid** [comment]
Merged Sparkle update, bumped the package 1.43, applied new package here.
Ready to merge @spolu!

**Pull request merged**